doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,48 @@ After programming and testing an application, you probably want to make some mod
 Adding files and changing compiler settings
 *******************************************
 
-All files that your application uses must be specified in the :file:`CMakeLists.txt` file.
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information about how the build system works in Zephyr, see :ref:`zephyr:build_overview` and :ref:`zephyr:application` pages in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr provides a CMake package that must be loaded by the application into its :file:`CMakeLists.txt` file.
+When loaded, the application is able to reference items provided by both Zephyr and the |NCS|.
+
+Loading Zephyr's `CMake`_ package creates the ``app`` CMake target.
+All files that your application uses can be added to this target.
+This can be done directly in the :file:`CMakeLists.txt` file or in the files that are sourced by it.
```

Comment:
"uses" is too vague. An application uses kernel source files and BT files etc.

Let's not explain add_subdirectory here, it is a CMake detail that is not necessary to understand at this point. So cut

"All files that your application uses can be added to this target.
This can be done directly in the :file:`CMakeLists.txt` file or in the files that are sourced by it."

and add

"Application source files can be added to this target from the application :file:`CMakeLists.txt` file."

Reply:
This is a nice suggestion, thanks!